doc/kernel/usermode/syscalls.rst

```diff
@@ -189,6 +189,26 @@ generated due to its repetitive nature and can be found in
 ``include/generated/syscall_macros.h``. It is created by
 ``scripts/gen_syscall_header.py``.
 
+The final layer is the invocation of the system call itself. All architectures
+implementing system calls must implement the seven inline functions
+:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`.  These
+functions marshal arguments into designated CPU registers and perform the
+necessary privilege elevation. In this layer all arguments are treated as an
+unsigned 32-bit type, and there is always a 32-bit unsigned return value, which
+may or may not be used.
+
+Some system calls may have more than six arguments. The number of arguments
+passed via registers is fixed at six for all architectures, additional
+arguments will need to be passed in a struct which needs to be treated as
+untrusted memory in the handler function. This is done by the derived
+functions :c:func:`_syscall_invoke7` through :c:func:`_syscall_invoke10`.
+
+Some system calls may return a value which will not fit in a 32-bit register,
+such as APIs that return a 64-bit value. In this scenario, the return value
+is populated in a memory buffer that is passed in as an argument, see the
```

Comment:
End the sentence and start a new one:  "as an argument.  For example, see the..."